vck/src/commonMain/kotlin/at/asitplus/wallet/lib/jws/JwsService.kt

```diff
@@ -142,14 +149,20 @@ class DefaultJwsService(private val cryptoService: CryptoService) : JwsService {
         addJsonWebKey: Boolean,
         addX5c: Boolean
     ): KmmResult<JwsSigned> = catching {
-        var copy = header?.copy(algorithm = cryptoService.keyPairAdapter.signingAlgorithm.toJwsAlgorithm().getOrThrow())
-            ?: JwsHeader(algorithm = cryptoService.keyPairAdapter.signingAlgorithm.toJwsAlgorithm().getOrThrow())
+        var copy = header?.copy(
+            algorithm = cryptoService.keyMaterial.signatureAlgorithm.toJwsAlgorithm().getOrThrow()
+        )
+            ?: JwsHeader(
+                algorithm = cryptoService.keyMaterial.signatureAlgorithm.toJwsAlgorithm()
+                    .getOrThrow()
+            )
         if (addKeyId)
-            copy = copy.copy(keyId = cryptoService.keyPairAdapter.jsonWebKey.keyId)
+            copy = copy.copy(keyId = cryptoService.keyMaterial.jsonWebKey.keyId)
         if (addJsonWebKey)
-            copy = copy.copy(jsonWebKey = cryptoService.keyPairAdapter.jsonWebKey)
+            copy = copy.copy(jsonWebKey = cryptoService.keyMaterial.jsonWebKey)
         if (addX5c)
-            copy = copy.copy(certificateChain = listOf(cryptoService.keyPairAdapter.certificate!!)) //TODO cleanup/nullchecks
+            copy =
+                copy.copy(certificateChain = listOf(cryptoService.keyMaterial.getCertificate()!!)) //TODO cleanup/nullchecks
```

Comment:
Is this TODO relevant now?

Reply:
it should go and it should throw inside this catching block, because this is an error. I changed the comment